[docs] Migrate docs from AsciiDoc to Markdown

eedugon opened the pull request:
Migrates docs from AsciiDoc to Markdown

This PR should substitute https://github.com/elastic/cloud-on-k8s/pull/8494 created by @colleenmcginnis (I didn't have permissions to merge the latest changes in main, and I might need to add a few extra changes to the content).

 > Note: When you get the build working and click the View deployment button, you'll need to add /reference to the end of the URL you're sent to in order to see a rendered page.

eedugon commented:
This PR cannot be merged until the CI jobs are adjusted (https://buildkite.com/elastic/cloud-on-k8s-operator/builds/10389#019547bc-f8e5-46dd-93a0-1ae3a0f48d99).

Based on @pebrc comments, the steps to take are:

1.  disable the master/main docs build in favour of the new CI checks merged on https://github.com/elastic/cloud-on-k8s/pull/8493
2. adjust the repository tooling (which will otherwise fail all builds)
3. merge this PR.

pebrc commented:
@eedugon can we get this into a mergeable state? As we are preparing for the 3.0 release and we need to make docs changes it would be good to have the docs as close as possible to the final file layout to avoid double effort on our end maintaining changes in two repos.

eedugon commented:
@pebrc , the build is failing because of the content you have auto-generated in `third-party-dependencies.md`.

Your links look like:
```
| [https://dario.cat/mergo]($$dario.cat/mergo$$) | v1.0.1 | BSD-3-Clause |
| [https://github.com/Masterminds/sprig]($$github.com/Masterminds/sprig/v3$$) | v3.3.0 | MIT |
| [https://github.com/blang/semver]($$github.com/blang/semver/v4$$) | v4.0.0 | MIT |
| [https://github.com/davecgh/go-spew]($$github.com/davecgh/go-spew$$) | v1.1.2-0.20180830191138-d8f796af33cc | ISC |
| [https://github.com/elastic/go-ucfg]($$github.com/elastic/go-ucfg$$) | v0.8.8 | Apache-2.0 |
| [https://github.com/ghodss/yaml]($$github.com/ghodss/yaml$$) | v1.0.0 | MIT |
```

And in the static content we had provided it looked like:

```
| [dario.cat/mergo](https://dario.cat/mergo) | v1.0.1 | BSD-3-Clause |
| [github.com/Masterminds/sprig/v3](https://github.com/Masterminds/sprig) | v3.3.0 | MIT |
| [github.com/blang/semver/v4](https://github.com/blang/semver) | v4.0.0 | MIT |
| [github.com/davecgh/go-spew](https://github.com/davecgh/go-spew) | v1.1.2-0.20180830191138-d8f796af33cc | ISC |
| [github.com/elastic/go-ucfg](https://github.com/elastic/go-ucfg) | v0.8.8 | Apache-2.0 |
| [github.com/ghodss/yaml](https://github.com/ghodss/yaml) | v1.0.0 | MIT |
| [github.com/gkampitakis/go-snaps](https://github.com/gkampitakis/go-snaps) | v0.5.8 | MIT |
```

I'm sorry I didn't get this in your previous PR. Would you please fix the auto-generated file in main?

As an extra issue, the markdown comments in the new Docs V3 system should be made with `%` (details [here](https://elastic.github.io/docs-builder/syntax/comments/) ), so the following won't work as a comment:

`<!--- Generated documentation. Please do not edit. --->`

It should be:

% Generated documentation. Please do not edit.

pebrc commented:
> @pebrc : FYI, while fixing the conflicts:
> 
>     * File `advanced-topics/service-meshes.asciidoc` was modified in main branch and we are deleting it in this PR.

I can open a PR against the new docs repo for that.


> 
>     * File `release-notes/2.16.0.asciidoc` was modified in main branch and we are deleting it in this PR.
> 

We can ignore this change as it was backported to the 2.16 branch already.

pebrc commented:
You are running into another problem with the build.

https://github.com/elastic/cloud-on-k8s/blob/95acd68d38d7fc446b8138313be366cdd34d5fac/Makefile#L499-L507

This is some code that we added to verify that the documentation was always up to date compared to the code. Unfortunately with the docs moving out of this repo this check won't work anymore and you will have to either remove it or move the documentation of the update predicates back into this repo as reference documenation.

eedugon commented:
working on this, @pebrc, thanks for flagging it.

eedugon commented:
@pebrc : This PR is mergeable now. Feel free to review it.

We have added to the reference documentation the doc about [upgrade predicates and the example use case](https://docs-v3-preview.elastic.dev/elastic/cloud-on-k8s/pull/8496/reference/upgrade-predicates)

In the main nodes orchestration doc we have added an introduction and link in the [Advanced control during rolling upgrades](https://docs-v3-preview.elastic.dev/elastic/docs-content/tree/main/deploy-manage/deploy/cloud-on-k8s/nodes-orchestration#k8s-advanced-upgrade-control) section.

Let us know your thoughts. We can also change the title of the new reference doc at a later stage.
